In [80]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [81]:
import pandas as pds
import matplotlib.pyplot as plt
import seaborn as sns

In [82]:
dset=pds.read_csv("/content/train.csv")
dset

Unnamed: 0                              Name    Location  Year  \
0              1  Hyundai Creta 1.6 CRDi SX Option        Pune  2015   
1              2                      Honda Jazz V     Chennai  2011   
2              3                 Maruti Ertiga VDI     Chennai  2012   
3              4   Audi A4 New 2.0 TDI Multitronic  Coimbatore  2013   
4              6            Nissan Micra Diesel XV      Jaipur  2013   
...          ...                               ...         ...   ...   
5842        6014                  Maruti Swift VDI       Delhi  2014   
5843        6015          Hyundai Xcent 1.1 CRDi S      Jaipur  2015   
5844        6016             Mahindra Xylo D4 BSIV      Jaipur  2012   
5845        6017                Maruti Wagon R VXI     Kolkata  2013   
5846        6018             Chevrolet Beat Diesel   Hyderabad  2011   

      Kilometers_Driven Fuel_Type Transmission Owner_Type     Mileage  \
0                 41000    Diesel       Manual      First  19.67 kmpl   
1                 46000    Petrol       Manual      First    13 km/kg   
2                 87000    Diesel       Manual      First  20.77 kmpl   
3                 40670    Diesel    Automatic     Second   15.2 kmpl   
4                 86999    Diesel       Manual      First  23.08 kmpl   
...                 ...       ...          ...        ...         ...   
5842              27365    Diesel       Manual      First   28.4 kmpl   
5843             100000    Diesel       Manual      First   24.4 kmpl   
5844              55000    Diesel       Manual     Second   14.0 kmpl   
5845              46000    Petrol       Manual      First   18.9 kmpl   
5846              47000    Diesel       Manual      First  25.44 kmpl   

       Engine      Power  Seats  New_Price  Price  
0     1582 CC  126.2 bhp    5.0        NaN  12.50  
1     1199 CC   88.7 bhp    5.0  8.61 Lakh   4.50  
2     1248 CC  88.76 bhp    7.0        NaN   6.00  
3     1968 CC  140.8 bhp    5.0        NaN  17.74  
4     1461 CC   63.1 bhp    5.0        NaN   3.50  
...       ...        ...    ...        ...    ...  
5842  1248 CC     74 bhp    5.0  7.88 Lakh   4.75  
5843  1120 CC     71 bhp    5.0        NaN   4.00  
5844  2498 CC    112 bhp    8.0        NaN   2.90  
5845   998 CC   67.1 bhp    5.0        NaN   2.65  
5846   936 CC   57.6 bhp    5.0        NaN   2.50  

[5847 rows x 14 columns]

1) Look for the missing values in all the columns and either impute them (replace with mean, median, or mode) or drop them. Justify your action for this task.

In [83]:
def extract_numeric(data_Set):
    return pds.to_numeric(data_Set.astype(str).str.extract(r'([\d.]+)')[0], errors='coerce')
dset_num=pds.DataFrame(columns=[])
dset_num["Mileage"]=extract_numeric(dset["Mileage"])
dset_num["Engine"]=extract_numeric(dset["Engine"])
dset_num["Power"]=extract_numeric(dset["Power"])
print(dset_num)
dset["Mileage"].fillna(dset_num["Mileage"].mean(), inplace=True)
dset["Engine"].fillna(dset_num["Engine"].mean(), inplace=True)
dset["Power"].fillna(dset_num["Power"].mean(), inplace=True)
dset["Seats"].fillna(dset["Seats"].mode()[0], inplace=True)
dset["Fuel_Type"].fillna(dset["Fuel_Type"].mode()[0], inplace=True)
dset["Transmission"].fillna(dset["Transmission"].mode()[0], inplace=True)
dset["Owner_Type"].fillna(dset["Owner_Type"].mode()[0], inplace=True)

dset.drop(columns=["New_Price"], inplace=True)
dset

      Mileage  Engine   Power
0       19.67  1582.0  126.20
1       13.00  1199.0   88.70
2       20.77  1248.0   88.76
3       15.20  1968.0  140.80
4       23.08  1461.0   63.10
...       ...     ...     ...
5842    28.40  1248.0   74.00
5843    24.40  1120.0   71.00
5844    14.00  2498.0  112.00
5845    18.90   998.0   67.10
5846    25.44   936.0   57.60

[5847 rows x 3 columns]


<ipython-input-83-254c1bc1ccd7>:8: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=True)', try using 'df.method({col: value}, inplace=True)' or df[col] = df[col].method(value) instead, to perform the operation inplace on the original object.


  dset["Mileage"].fillna(dset_num["Mileage"].mean(), inplace=True)
<ipython-input-83-254c1bc1ccd7>:9: FutureWarning: A value is trying to be set on a copy of a DataFrame or Series through chained assignment using an inplace method.
The behavior will change in pandas 3.0. This inplace method will never work because the intermediate object on which we are setting values always behaves as a copy.

For example, when doing 'df[col].method(value, inplace=

Unnamed: 0                              Name    Location  Year  \
0              1  Hyundai Creta 1.6 CRDi SX Option        Pune  2015   
1              2                      Honda Jazz V     Chennai  2011   
2              3                 Maruti Ertiga VDI     Chennai  2012   
3              4   Audi A4 New 2.0 TDI Multitronic  Coimbatore  2013   
4              6            Nissan Micra Diesel XV      Jaipur  2013   
...          ...                               ...         ...   ...   
5842        6014                  Maruti Swift VDI       Delhi  2014   
5843        6015          Hyundai Xcent 1.1 CRDi S      Jaipur  2015   
5844        6016             Mahindra Xylo D4 BSIV      Jaipur  2012   
5845        6017                Maruti Wagon R VXI     Kolkata  2013   
5846        6018             Chevrolet Beat Diesel   Hyderabad  2011   

      Kilometers_Driven Fuel_Type Transmission Owner_Type     Mileage  \
0                 41000    Diesel       Manual      First  19.67 kmpl   
1                 46000    Petrol       Manual      First    13 km/kg   
2                 87000    Diesel       Manual      First  20.77 kmpl   
3                 40670    Diesel    Automatic     Second   15.2 kmpl   
4                 86999    Diesel       Manual      First  23.08 kmpl   
...                 ...       ...          ...        ...         ...   
5842              27365    Diesel       Manual      First   28.4 kmpl   
5843             100000    Diesel       Manual      First   24.4 kmpl   
5844              55000    Diesel       Manual     Second   14.0 kmpl   
5845              46000    Petrol       Manual      First   18.9 kmpl   
5846              47000    Diesel       Manual      First  25.44 kmpl   

       Engine      Power  Seats  Price  
0     1582 CC  126.2 bhp    5.0  12.50  
1     1199 CC   88.7 bhp    5.0   4.50  
2     1248 CC  88.76 bhp    7.0   6.00  
3     1968 CC  140.8 bhp    5.0  17.74  
4     1461 CC   63.1 bhp    5.0   3.50  
...       ...        ...    ...    ...  
5842  1248 CC     74 bhp    5.0   4.75  
5843  1120 CC     71 bhp    5.0   4.00  
5844  2498 CC    112 bhp    8.0   2.90  
5845   998 CC   67.1 bhp    5.0   2.65  
5846   936 CC   57.6 bhp    5.0   2.50  

[5847 rows x 13 columns]

**Justity**: All the Numerice Data Type colummn null values have been replaced with its mean. Categorical values are replaced with their columns mode value. New_Price is a numeric variable and the data set have only 815 entries for the "New_Price" column out of 5847, which is mostly null. Also prices vary greatly and often aren't avaliable for older models. So, the "New_Price" column is dropped off the dataset.

2) Remove the units from some of the attributes and only keep the numerical values (for example remove kmpl from “Mileage”, CC from “Engine”, bhp from “Power”, and lakh from “New_price”).

In [84]:
dset["Mileage"] = dset["Mileage"].astype(str).str.extract(r'([\d\.]+)').astype(float)
dset["Engine"] = dset["Engine"].astype(str).str.extract(r'([\d\.]+)').astype(float)
dset["Power"] = dset["Power"].astype(str).str.extract(r'([\d\.]+)').astype(float)
#dset["New_Price"] = dset["New_Price"].astype(str).str.extract(r'([\d\.]+)').astype(float)
dset

Unnamed: 0                              Name    Location  Year  \
0              1  Hyundai Creta 1.6 CRDi SX Option        Pune  2015   
1              2                      Honda Jazz V     Chennai  2011   
2              3                 Maruti Ertiga VDI     Chennai  2012   
3              4   Audi A4 New 2.0 TDI Multitronic  Coimbatore  2013   
4              6            Nissan Micra Diesel XV      Jaipur  2013   
...          ...                               ...         ...   ...   
5842        6014                  Maruti Swift VDI       Delhi  2014   
5843        6015          Hyundai Xcent 1.1 CRDi S      Jaipur  2015   
5844        6016             Mahindra Xylo D4 BSIV      Jaipur  2012   
5845        6017                Maruti Wagon R VXI     Kolkata  2013   
5846        6018             Chevrolet Beat Diesel   Hyderabad  2011   

      Kilometers_Driven Fuel_Type Transmission Owner_Type  Mileage  Engine  \
0                 41000    Diesel       Manual      First    19.67  1582.0   
1                 46000    Petrol       Manual      First    13.00  1199.0   
2                 87000    Diesel       Manual      First    20.77  1248.0   
3                 40670    Diesel    Automatic     Second    15.20  1968.0   
4                 86999    Diesel       Manual      First    23.08  1461.0   
...                 ...       ...          ...        ...      ...     ...   
5842              27365    Diesel       Manual      First    28.40  1248.0   
5843             100000    Diesel       Manual      First    24.40  1120.0   
5844              55000    Diesel       Manual     Second    14.00  2498.0   
5845              46000    Petrol       Manual      First    18.90   998.0   
5846              47000    Diesel       Manual      First    25.44   936.0   

       Power  Seats  Price  
0     126.20    5.0  12.50  
1      88.70    5.0   4.50  
2      88.76    7.0   6.00  
3     140.80    5.0  17.74  
4      63.10    5.0   3.50  
...      ...    ...    ...  
5842   74.00    5.0   4.75  
5843   71.00    5.0   4.00  
5844  112.00    8.0   2.90  
5845   67.10    5.0   2.65  
5846   57.60    5.0   2.50  

[5847 rows x 13 columns]

3) Change the categorical variables (“Fuel_Type” and “Transmission”) into numerical one hot encoded value.

If the Fuel_type is "Petrol" then the values will be 1, and 2 for "Diesel" Fuel_type

If the Transmission type is "Manual" it's give value 1, and 2 for the Transmission type "Automatic"

In [85]:
dset["Fuel_Type"] = dset["Fuel_Type"].apply(lambda x: 1 if x.lower()=="petrol" else 2)
dset["Transmission"] = dset["Transmission"].apply(lambda x: 1 if x.lower()=="manual" else 2)
dset

Unnamed: 0                              Name    Location  Year  \
0              1  Hyundai Creta 1.6 CRDi SX Option        Pune  2015   
1              2                      Honda Jazz V     Chennai  2011   
2              3                 Maruti Ertiga VDI     Chennai  2012   
3              4   Audi A4 New 2.0 TDI Multitronic  Coimbatore  2013   
4              6            Nissan Micra Diesel XV      Jaipur  2013   
...          ...                               ...         ...   ...   
5842        6014                  Maruti Swift VDI       Delhi  2014   
5843        6015          Hyundai Xcent 1.1 CRDi S      Jaipur  2015   
5844        6016             Mahindra Xylo D4 BSIV      Jaipur  2012   
5845        6017                Maruti Wagon R VXI     Kolkata  2013   
5846        6018             Chevrolet Beat Diesel   Hyderabad  2011   

      Kilometers_Driven  Fuel_Type  Transmission Owner_Type  Mileage  Engine  \
0                 41000          2             1      First    19.67  1582.0   
1                 46000          1             1      First    13.00  1199.0   
2                 87000          2             1      First    20.77  1248.0   
3                 40670          2             2     Second    15.20  1968.0   
4                 86999          2             1      First    23.08  1461.0   
...                 ...        ...           ...        ...      ...     ...   
5842              27365          2             1      First    28.40  1248.0   
5843             100000          2             1      First    24.40  1120.0   
5844              55000          2             1     Second    14.00  2498.0   
5845              46000          1             1      First    18.90   998.0   
5846              47000          2             1      First    25.44   936.0   

       Power  Seats  Price  
0     126.20    5.0  12.50  
1      88.70    5.0   4.50  
2      88.76    7.0   6.00  
3     140.80    5.0  17.74  
4      63.10    5.0   3.50  
...      ...    ...    ...  
5842   74.00    5.0   4.75  
5843   71.00    5.0   4.00  
5844  112.00    8.0   2.90  
5845   67.10    5.0   2.65  
5846   57.60    5.0   2.50  

[5847 rows x 13 columns]

 4) Create one more feature and add this column to the dataset (you can use mutate function in R for this). For example, you can calculate the current age of the car by subtracting “Year” value from the current year.

In [89]:
dset["Car_Age"]=2025-dset["Year"]
dset

Unnamed: 0                              Name    Location  Year  \
0              1  Hyundai Creta 1.6 CRDi SX Option        Pune  2015   
1              2                      Honda Jazz V     Chennai  2011   
2              3                 Maruti Ertiga VDI     Chennai  2012   
3              4   Audi A4 New 2.0 TDI Multitronic  Coimbatore  2013   
4              6            Nissan Micra Diesel XV      Jaipur  2013   
...          ...                               ...         ...   ...   
5842        6014                  Maruti Swift VDI       Delhi  2014   
5843        6015          Hyundai Xcent 1.1 CRDi S      Jaipur  2015   
5844        6016             Mahindra Xylo D4 BSIV      Jaipur  2012   
5845        6017                Maruti Wagon R VXI     Kolkata  2013   
5846        6018             Chevrolet Beat Diesel   Hyderabad  2011   

      Kilometers_Driven  Fuel_Type  Transmission Owner_Type  Mileage  Engine  \
0                 41000          2             1      First    19.67  1582.0   
1                 46000          1             1      First    13.00  1199.0   
2                 87000          2             1      First    20.77  1248.0   
3                 40670          2             2     Second    15.20  1968.0   
4                 86999          2             1      First    23.08  1461.0   
...                 ...        ...           ...        ...      ...     ...   
5842              27365          2             1      First    28.40  1248.0   
5843             100000          2             1      First    24.40  1120.0   
5844              55000          2             1     Second    14.00  2498.0   
5845              46000          1             1      First    18.90   998.0   
5846              47000          2             1      First    25.44   936.0   

       Power  Seats  Price  Car_Age  
0     126.20    5.0  12.50       10  
1      88.70    5.0   4.50       14  
2      88.76    7.0   6.00       13  
3     140.80    5.0  17.74       12  
4      63.10    5.0   3.50       12  
...      ...    ...    ...      ...  
5842   74.00    5.0   4.75       11  
5843   71.00    5.0   4.00       10  
5844  112.00    8.0   2.90       13  
5845   67.10    5.0   2.65       12  
5846   57.60    5.0   2.50       14  

[5847 rows x 14 columns]

 5) Perform select, filter, rename, mutate, arrange and summarize with group by operations (or their equivalent operations in python) on this dataset.

**SELECT:**

In [90]:
dset_select=dset[["Name","Year","Price","Fuel_Type","Transmission"]]
dset_select

Name  Year  Price  Fuel_Type  Transmission
0     Hyundai Creta 1.6 CRDi SX Option  2015  12.50          2             1
1                         Honda Jazz V  2011   4.50          1             1
2                    Maruti Ertiga VDI  2012   6.00          2             1
3      Audi A4 New 2.0 TDI Multitronic  2013  17.74          2             2
4               Nissan Micra Diesel XV  2013   3.50          2             1
...                                ...   ...    ...        ...           ...
5842                  Maruti Swift VDI  2014   4.75          2             1
5843          Hyundai Xcent 1.1 CRDi S  2015   4.00          2             1
5844             Mahindra Xylo D4 BSIV  2012   2.90          2             1
5845                Maruti Wagon R VXI  2013   2.65          1             1
5846             Chevrolet Beat Diesel  2011   2.50          2             1

[5847 rows x 5 columns]

**FILTER:**

In [91]:
dset_filter=dset[(dset["Transmission"]==1)&(dset["Fuel_Type"]==2)]
dset_filter

Unnamed: 0                                 Name    Location  Year  \
0              1     Hyundai Creta 1.6 CRDi SX Option        Pune  2015   
2              3                    Maruti Ertiga VDI     Chennai  2012   
4              6               Nissan Micra Diesel XV      Jaipur  2013   
6              8  Volkswagen Vento Diesel Comfortline        Pune  2013   
7              9       Tata Indica Vista Quadrajet LS     Chennai  2012   
...          ...                                  ...         ...   ...   
5841        6013                Honda Amaze VX i-DTEC  Coimbatore  2015   
5842        6014                     Maruti Swift VDI       Delhi  2014   
5843        6015             Hyundai Xcent 1.1 CRDi S      Jaipur  2015   
5844        6016                Mahindra Xylo D4 BSIV      Jaipur  2012   
5846        6018                Chevrolet Beat Diesel   Hyderabad  2011   

      Kilometers_Driven  Fuel_Type  Transmission Owner_Type  Mileage  Engine  \
0                 41000          2             1      First    19.67  1582.0   
2                 87000          2             1      First    20.77  1248.0   
4                 86999          2             1      First    23.08  1461.0   
6                 64430          2             1      First    20.54  1598.0   
7                 65932          2             1     Second    22.30  1248.0   
...                 ...        ...           ...        ...      ...     ...   
5841              70602          2             1      First    25.80  1498.0   
5842              27365          2             1      First    28.40  1248.0   
5843             100000          2             1      First    24.40  1120.0   
5844              55000          2             1     Second    14.00  2498.0   
5846              47000          2             1      First    25.44   936.0   

       Power  Seats  Price  Car_Age  
0     126.20    5.0  12.50       10  
2      88.76    7.0   6.00       13  
4      63.10    5.0   3.50       12  
6     103.60    5.0   5.20       12  
7      74.00    5.0   1.95       13  
...      ...    ...    ...      ...  
5841   98.60    5.0   4.83       10  
5842   74.00    5.0   4.75       11  
5843   71.00    5.0   4.00       10  
5844  112.00    8.0   2.90       13  
5846   57.60    5.0   2.50       14  

[2055 rows x 14 columns]

**RENAME:**

In [92]:
dset_rename=dset.rename(columns={"Kilometers_Driven":"Distance_Travelled"})
dset_rename

Unnamed: 0                              Name    Location  Year  \
0              1  Hyundai Creta 1.6 CRDi SX Option        Pune  2015   
1              2                      Honda Jazz V     Chennai  2011   
2              3                 Maruti Ertiga VDI     Chennai  2012   
3              4   Audi A4 New 2.0 TDI Multitronic  Coimbatore  2013   
4              6            Nissan Micra Diesel XV      Jaipur  2013   
...          ...                               ...         ...   ...   
5842        6014                  Maruti Swift VDI       Delhi  2014   
5843        6015          Hyundai Xcent 1.1 CRDi S      Jaipur  2015   
5844        6016             Mahindra Xylo D4 BSIV      Jaipur  2012   
5845        6017                Maruti Wagon R VXI     Kolkata  2013   
5846        6018             Chevrolet Beat Diesel   Hyderabad  2011   

      Distance_Travelled  Fuel_Type  Transmission Owner_Type  Mileage  Engine  \
0                  41000          2             1      First    19.67  1582.0   
1                  46000          1             1      First    13.00  1199.0   
2                  87000          2             1      First    20.77  1248.0   
3                  40670          2             2     Second    15.20  1968.0   
4                  86999          2             1      First    23.08  1461.0   
...                  ...        ...           ...        ...      ...     ...   
5842               27365          2             1      First    28.40  1248.0   
5843              100000          2             1      First    24.40  1120.0   
5844               55000          2             1     Second    14.00  2498.0   
5845               46000          1             1      First    18.90   998.0   
5846               47000          2             1      First    25.44   936.0   

       Power  Seats  Price  Car_Age  
0     126.20    5.0  12.50       10  
1      88.70    5.0   4.50       14  
2      88.76    7.0   6.00       13  
3     140.80    5.0  17.74       12  
4      63.10    5.0   3.50       12  
...      ...    ...    ...      ...  
5842   74.00    5.0   4.75       11  
5843   71.00    5.0   4.00       10  
5844  112.00    8.0   2.90       13  
5845   67.10    5.0   2.65       12  
5846   57.60    5.0   2.50       14  

[5847 rows x 14 columns]

**MUTATE:**

In [93]:
dset_filter["Car_Age"]=2025-dset_filter["Year"]
dset_filter

<ipython-input-93-0f1e61253e48>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  dset_filter["Car_Age"]=2025-dset_filter["Year"]


Unnamed: 0                                 Name    Location  Year  \
0              1     Hyundai Creta 1.6 CRDi SX Option        Pune  2015   
2              3                    Maruti Ertiga VDI     Chennai  2012   
4              6               Nissan Micra Diesel XV      Jaipur  2013   
6              8  Volkswagen Vento Diesel Comfortline        Pune  2013   
7              9       Tata Indica Vista Quadrajet LS     Chennai  2012   
...          ...                                  ...         ...   ...   
5841        6013                Honda Amaze VX i-DTEC  Coimbatore  2015   
5842        6014                     Maruti Swift VDI       Delhi  2014   
5843        6015             Hyundai Xcent 1.1 CRDi S      Jaipur  2015   
5844        6016                Mahindra Xylo D4 BSIV      Jaipur  2012   
5846        6018                Chevrolet Beat Diesel   Hyderabad  2011   

      Kilometers_Driven  Fuel_Type  Transmission Owner_Type  Mileage  Engine  \
0                 41000          2             1      First    19.67  1582.0   
2                 87000          2             1      First    20.77  1248.0   
4                 86999          2             1      First    23.08  1461.0   
6                 64430          2             1      First    20.54  1598.0   
7                 65932          2             1     Second    22.30  1248.0   
...                 ...        ...           ...        ...      ...     ...   
5841              70602          2             1      First    25.80  1498.0   
5842              27365          2             1      First    28.40  1248.0   
5843             100000          2             1      First    24.40  1120.0   
5844              55000          2             1     Second    14.00  2498.0   
5846              47000          2             1      First    25.44   936.0   

       Power  Seats  Price  Car_Age  
0     126.20    5.0  12.50       10  
2      88.76    7.0   6.00       13  
4      63.10    5.0   3.50       12  
6     103.60    5.0   5.20       12  
7      74.00    5.0   1.95       13  
...      ...    ...    ...      ...  
5841   98.60    5.0   4.83       10  
5842   74.00    5.0   4.75       11  
5843   71.00    5.0   4.00       10  
5844  112.00    8.0   2.90       13  
5846   57.60    5.0   2.50       14  

[2055 rows x 14 columns]

**ARRANGE:**

In [94]:
dset_sorted=dset.sort_values(by="Kilometers_Driven", ascending=False)
dset_sorted

Unnamed: 0                                        Name Location  Year  \
2259        2328                   BMW X5 xDrive 30d M Sport  Chennai  2017   
326          340      Skoda Octavia Ambition Plus 2.0 TDI AT  Kolkata  2013   
1805        1860            Volkswagen Vento Diesel Highline  Chennai  2013   
344          358                       Hyundai i10 Magna 1.2  Chennai  2009   
2997        3092                        Honda City i VTEC SV  Kolkata  2015   
...          ...                                         ...      ...   ...   
1547        1598                        Tata Xenon XT EX 4X2   Jaipur  2017   
5772        5941                            Renault KWID RXL     Pune  2018   
2558        2646  Ford Figo 2015-2019 1.2P Sports Edition MT   Mumbai  2019   
5451        5606                  Maruti Wagon R ZXI AMT 1.2   Mumbai  2019   
1315        1361               Maruti Alto 800 2016-2019 VXI   Mumbai  2019   

      Kilometers_Driven  Fuel_Type  Transmission Owner_Type  Mileage  Engine  \
2259            6500000          2             2      First    15.97  2993.0   
326              775000          2             2      First    19.30  1968.0   
1805             720000          2             1      First    20.54  1598.0   
344              620000          1             1      First    20.36  1197.0   
2997             480000          1             1      First    17.40  1497.0   
...                 ...        ...           ...        ...      ...     ...   
1547               1000          2             1      First    13.49  2179.0   
5772               1000          1             1      First    25.17   799.0   
2558               1000          1             1      First    18.12  1196.0   
5451                600          1             2      First    21.50  1197.0   
1315                171          1             1      First    24.70   796.0   

       Power  Seats  Price  Car_Age  
2259  258.00    5.0  65.00        8  
326   141.00    5.0   7.50       12  
1805  103.60    5.0   5.90       12  
344    78.90    5.0   2.70       16  
2997  117.30    5.0   5.00       10  
...      ...    ...    ...      ...  
1547  138.03    5.0   7.50        8  
5772   53.30    5.0   3.40        7  
2558   86.80    5.0   6.80        6  
5451   81.80    5.0   6.25        6  
1315   47.30    5.0   3.60        6  

[5847 rows x 14 columns]

**SUMMARIZE WITH GROUP_BY:**

In [95]:
df_summary=dset_sorted.groupby("Transmission")["Price"].mean().reset_index()
df_summary.rename(columns={"price": "Avg_Selling_Price"}, inplace=True)
df_summary

Transmission      Price
0             1   5.415998
1             2  19.889182